### PR TITLE
Unify search and package listing.

### PR DESCRIPTION
**Pull request by isoos**
There are two minor follow-up tasks with the URL handling (added TODO for them), but I felt doing them in the same PR will make it too bloated.

### CODE REVIEW

File: app/lib/frontend/handlers.dart [hunk 1]
```diff
@@ -446,19 +435,30 @@ Future<shelf.Response> _packagesHandlerHtmlV2(
     shelf.Request request, String platform) async {
   final int page = _pageFromUrl(request.url);
   final int offset = PackageLinks.resultsPerPage * (page - 1);
+  final queryText = request.requestedUri.queryParameters['q'] ?? '';
+  // TODO: handle package-prefix
+
+  // TODO: handle sort order
```
Comment (kevmoo) [inline app/lib/frontend/handlers.dart]: Yeah, this makes this pretty broken

File: app/test/frontend/golden/v2/web_landing_page.html [hunk 2]
```diff
@@ -60,9 +60,8 @@ <h1 class="_visuallyhidden">Dart pub</h1>
     <main class="home-banner">
       <h2 class="_visuallyhidden">Dart package manager</h2><img class="logo" src="/static/v2/img/site-logo.svg" alt="Dart package manager">
       <p class="text">Find and use packages to build <a href="/experimental/flutter">Flutter</a>, <a href="/experimental/web">web</a> and <a href="/experimental/server">server</a> apps with <a target="_blank" href="https://www.dartlang.org">Dart</a>.</p>
-      <form class="search-bar" action="/experimental/search">
+      <form class="search-bar" action="/experimental/server/packages">
```
Comment (kevmoo) [inline app/test/frontend/golden/v2/web_landing_page.html]: shouldn't the web landing page point to web packages?

File: app/test/frontend/golden/v2/pkg_show_page_flutter_plugin.html [hunk 3]
```diff
@@ -59,9 +59,8 @@ <h1 class="_visuallyhidden">Dart pub</h1>
 
   <div class="_banner-bg">
     <main class="package-banner">
-      <form class="search-bar" action="/experimental/search">
+      <form class="search-bar" action="/experimental/packages">
```
Comment (kevmoo) [inline app/test/frontend/golden/v2/pkg_show_page_flutter_plugin.html]: /experimental/flutter/packages?

File: app/lib/frontend/handlers.dart (hunk 1 above)
Comment (isoos) [reply app/lib/frontend/handlers.dart]: Yeah, the default order is context-dependent.

File: app/test/frontend/golden/v2/pkg_show_page_flutter_plugin.html (hunk 3 above)
Comment (isoos) [reply app/test/frontend/golden/v2/pkg_show_page_flutter_plugin.html]: This is a package show page, and we don't have good platform to chose (especially if it is not flutter-only).

File: app/test/frontend/golden/v2/web_landing_page.html (hunk 2 above)
Comment (isoos) [reply app/test/frontend/golden/v2/web_landing_page.html]: good catch, bad input in test, fixing.

**Pull request merged**